### PR TITLE
Create tagged-releases.md

### PR DESCRIPTION
**Pull request by loganmcdonald-noaa**
Draft of information about tagged releases.

### CODE REVIEW

File: docs/product/tagged-releases.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+Click "create new tag" - it will take you to a place to write release notes. Name the release the same as the tag. Then release! You can watch the steps here: https://github.com/weather-gov/weather.gov/actions/workflows/deploy-beta.yaml
+
+## Release cheklist
+
+As part of a release you may follow the following checklist: 
+
+- [ ] Compare [staging](https://weathergov-staging.app.cloud.gov/) and [beta](https://beta.weather.gov/) for differences in configuration, content, and functionality
+- [ ] Look at [commits to main](https://github.com/weather-gov/weather.gov/commits/main) and check what has been merged since the previous release
+- [ ] Write comprehensive release notes making sure to reference relevant pull requests
```
Comment (colinmurphy01) [inline docs/product/tagged-releases.md:19]: Is comprehensive just:

- What's changed
- What's new

File: docs/product/tagged-releases.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+- [ ] Do an accessibility review of what is on staging if anything functional or visual has been changed
```
Comment (colinmurphy01) [inline docs/product/tagged-releases.md:20]: How thorough of an accessibility review should be conducted? Just [our checklist](https://github.com/weather-gov/weather.gov/blob/main/docs/code-review-templates/code-review-web.md#accessibility)?

**Review by colinmurphy01 — APPROVED** [review]
Some clarifications would be helpful, but otherwise this is great. Thank you for putting this together.

File: docs/product/tagged-releases.md (hunk 2 above)
Comment (greg-does-weather) [reply docs/product/tagged-releases.md:20]: @colinmurphy01 This is a product question. 😉

File: docs/product/tagged-releases.md (hunk 1 above)
Comment (greg-does-weather) [reply docs/product/tagged-releases.md:19]: Comprehensive is however much is appropriate to meet product needs. 😉

File: docs/product/tagged-releases.md (hunk 2 above)
Comment (colinmurphy01) [reply docs/product/tagged-releases.md:20]:
```suggestion
- [ ] Run through [the manual accessibility checklist](https://github.com/weather-gov/weather.gov/blob/main/docs/code-review-templates/code-review-web.md#accessibility) of what is on staging if anything functional or visual has been changed
```

**Pull request merged**